### PR TITLE
feat(handlers): add upsertSlashCommands()

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**
Reference: 
image

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no
  code changes

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed,
    parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation,
  README, etc.

### CODE REVIEW

File: src/api/handlers/webhook.ts [hunk 1]
```diff
@@ -270,6 +271,42 @@ export function upsertSlashCommand(
...
+      throw new Error(Errors.INVALID_SLASH_NAME);
+    }
+
+    if (
+      [...option.description].length < 1 || [...option.description].length > 100
+    ) {
+      throw new Error(Errors.INVALID_SLASH_DESCRIPTION);
+    }
+
+    option["id"] = option.commandID;
```
Comment (Skillz4Killz) [inline src/api/handlers/webhook.ts:295]: What is this for?

File: src/types/webhook.ts [hunk 2]
```diff
@@ -228,3 +228,15 @@ export interface UpsertSlashCommandOptions {
...
+export interface UpsertSlashCommandsOptions {
+  commandID?: string;
+  /** 3-32 character command name */
+  name: string;
+  /** 1-100 character description */
+  description: string;
+  /** The parameters for the command */
+  options?: SlashCommandOption[];
+  /** @ignore this Parameter is used internally to make the request so you can safely ignore that. */
+  id?: string;
```
Comment (Skillz4Killz) [inline src/types/webhook.ts:241]: If i understood this right, whats the purpose of commandID? Why not just use `id`

Comment (ayntee) [reply src/types/webhook.ts:241]: To avoid ambiguity, I guess

Comment (itohatweb) [reply src/types/webhook.ts:241]: I want consistent typing. I have added this param so I don't need to use `@ts-ignore` but I can remove this param and add the comment if you want.

File: src/api/handlers/webhook.ts (hunk 1 above)
Comment (itohatweb) [reply src/api/handlers/webhook.ts:295]: The param discord takes for the commandID is just named id for the request so it needs to be set here.

**Review by ayntee — APPROVED** [review]
LGTM

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**